modules/http4s/src/smithy4s/http4s/internals/SmithyHttp4sClientEndpoint.scala

```diff
@@ -202,7 +203,10 @@ private[smithy4s] class SmithyHttp4sClientEndpointImpl[F[_], Op[_, _, _, _, _],
   ): F[Either[MetadataError, E]] = {
     val schema = oneOf.instance
     val errorMetadataDecoder = Metadata.PartialDecoder.fromSchema(schema)
-    implicit val errorCodec = entityCompiler.compilePartialEntityDecoder(schema)
+    implicit val errorCodec: EntityDecoder[F, BodyPartial[ErrorType]] =
+      entityCompiler.compilePartialEntityDecoder(
+        schema.addHints(InputOutput.Output.widen)
```

Comment:
Can we add this in the codegen instead? That's how we do it for input/output types of the operation. The upside is that it's a single place that reaches every interpreter.

(also wondering if `DumpModel` does this so that dynamic models see that trait)